jlibra-core/src/main/java/dev/jlibra/faucet/Faucet.java

```diff
@@ -0,0 +1,85 @@
+package dev.jlibra.faucet;
+
+import static org.apache.http.HttpHeaders.USER_AGENT;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.message.BasicHeader;
+
+import dev.jlibra.AuthenticationKey;
+import dev.jlibra.LibraRuntimeException;
+
+public class Faucet {
+
+    private CloseableHttpClient httpClient;
+
+    private String url;
+
+    private Faucet(CloseableHttpClient httpClient, String url) {
+        this.httpClient = httpClient;
+        this.url = url;
+    }
+
+    public void mint(AuthenticationKey authenticationKey, long amountInMicroLibras) {
+        try {
+            URIBuilder builder = new URIBuilder(url)
+                    .setParameter("amount", Long.toString(amountInMicroLibras))
+                    .setParameter("auth_key", authenticationKey.toString());
+            HttpPost httpPost = new HttpPost(builder.build().toString());
+            httpPost.addHeader(new BasicHeader(USER_AGENT, "JLibra"));
+            httpPost.setEntity(new StringEntity(""));
```

Comment:
is this empty StringEntity really necessary?

Reply:
Yes, it will throw an exception if it's left out. The weird thing about that api is that it's a POST but with an empty body and parameters are given as query parameters in the url.